qldpc/stim/noise_model.py

```diff
@@ -1,238 +1,919 @@
+                Clifford gates. Defaults to 0.0.
+            clifford_2q_depolarization: Depolarization probability for two-qubit
+                Clifford gates. Defaults to 0.0.
+            measure_flip_x: Probability of measurement result bit flip for X-basis
+                measurements. Defaults to 0.0.
+            measure_flip_y: Probability of measurement result bit flip for Y-basis
+                measurements. Defaults to 0.0.
+            measure_flip_z: Probability of measurement result bit flip for Z-basis
+                measurements. Defaults to 0.0.
+            measure_flip_xx: Probability of measurement result bit flip for XX-basis
+                joint measurements. Defaults to 0.0.
+            measure_flip_yy: Probability of measurement result bit flip for YY-basis
+                joint measurements. Defaults to 0.0.
+            measure_flip_zz: Probability of measurement result bit flip for ZZ-basis
+                joint measurements. Defaults to 0.0.
+            reset_x: Probability of bit flip error after RX (reset to |+⟩) operations.
+                Defaults to 0.0.
+            reset_y: Probability of bit flip error after RY (reset to |i⟩) operations.
+                Defaults to 0.0.
+            reset_z: Probability of bit flip error after R/RZ (reset to |0⟩) operations.
+                Defaults to 0.0.
+
+        Returns:
+            A NoiseModel instance configured with the specified error rates.
         """
-        Applies measurement operations and associated noise to the circuit.
+        return NoiseModel(
+            idle_depolarization=idle_depolarization,
+            additional_depolarization_waiting_for_m_or_r=additional_depolarization_waiting_for_m_or_r,
+            any_clifford_1q_rule=(
+                NoiseRule(after={"DEPOLARIZE1": clifford_1q_depolarization})
+                if clifford_1q_depolarization
+                else None
+            ),
+            any_clifford_2q_rule=(
+                NoiseRule(after={"DEPOLARIZE2": clifford_2q_depolarization})
+                if clifford_2q_depolarization
+                else None
+            ),
+            measure_rules={
+                "X": NoiseRule(after={}, flip_result=measure_flip_x),
+                "Y": NoiseRule(after={}, flip_result=measure_flip_y),
+                "Z": NoiseRule(after={}, flip_result=measure_flip_z),
+                "XX": NoiseRule(after={}, flip_result=measure_flip_xx),
+                "YY": NoiseRule(after={}, flip_result=measure_flip_yy),
+                "ZZ": NoiseRule(after={}, flip_result=measure_flip_zz),
+            },
+            gate_rules={
+                "RX": (NoiseRule(after={"Z_ERROR": reset_x}) if reset_x else NoiseRule(after={})),
+                "RY": (NoiseRule(after={"X_ERROR": reset_y}) if reset_y else NoiseRule(after={})),
+                "R": (NoiseRule(after={"X_ERROR": reset_z}) if reset_z else NoiseRule(after={})),
+            },
```

Comment:
Can reset operations use something like `reset_rules` instead of `gate_rules`? From the class description `gate_rules` should be used for overriding other noise rules.

Reply:
I'm confused about the request here.  What's wrong with calling (say) `R` a "gate" and using `gate_rules` for it?

Maybe the confusion here is coming from the fact that `gate_rules` is being used here to cover cases that aren't already covered by `any_clifford_1q_rule` and `any_clifford_2q_rule`.

Reply:
In fact, I think I want to go the other way: I don't see any reason to split `gate_rules` and `measure_rules`.  I think we should just have `rules`.

Reply:
Totally reasonable.  I'll work on refactoring.

Reply:
Resolving because this is no longer applicable.